dasharo-compatibility/basic-interfaces.robot

```diff
@@ -0,0 +1,43 @@
+*** Settings ***
+Library             Collections
+Library             OperatingSystem
+Library             Process
+Library             String
+Library             Telnet    timeout=30 seconds    connection_timeout=120 seconds
+Library             SSHLibrary    timeout=90 seconds
+Library             RequestsLibrary
+# TODO: maybe have a single file to include if we need to include the same
+# stuff in all test cases
+Resource            ../sonoff-rest-api/sonoff-api.robot
+Resource            ../rtectrl-rest-api/rtectrl.robot
+Resource            ../variables.robot
+Resource            ../keywords.robot
+Resource            ../keys.robot
+Resource            ../pikvm-rest-api/pikvm_comm.robot
+
+# TODO:
+# - document which setup/teardown keywords to use and what are they doing
+# - go threough them and make sure they are doing what the name suggest (not
+# exactly the case right now)
+Suite Setup         Run Keyword
+...                     Prepare Test Suite
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+BI001.001 Check serial output
```

Comment:
Cannot we simply do an Enter TianoCore Setup test?

Reply:
We do not have Dasharo firmware here (yet).

I proposed that we could also have a test checking operation of control interfaces, despite the firmware/payload type.

We could use such one to verify that platform was setup correctly.

Reply:
Yes, but the official firmware is TianoCore based...

Reply:
If this is the case here, maybe we can see if the `enter boot menu` tests pass as well on this platform.
I am not sure, though, if having this basic generic test at the same time is a problem.

Reply:
I downloaded the firmware from [intel.com](https://www.intel.com/content/www/us/en/download/674533/minnowboard-max-and-minnowboard-turbot-firmware-flash-images.html?wapkw=minnowboard%20turbot) and I can't enter setup. Instead, a lot of logs are printed, including this:
```
[Bds]=============Begin Load Options Dumping ...=============
  Driver Options:
  SysPrep Options:
  Boot Options:
    Boot0000: UEFI BootManagerMenuApp            0x0109
    Boot0001: Internal UEFI Shell 2.0            0x0001
    Boot0002: Enter Setup                0x0109
    Boot0003: UEFI UiApp                 0x0001
    Boot0004: UEFI PXEv4 (MAC:0008A20CB8BF)              0x0001
    Boot0005: UEFI PXEv6 (MAC:0008A20CB8BF)              0x0001
    Boot0006: UMnpSyncSendPacket: No network cable detected.
EFI HTTPv4 (MAC:0008A20CB8BF)            0x0001
    Boot0007: UEFI HTTPv6 (MAC:0008A20CB8BF)             0x0001
  PlatformRecovery Options:
    PlatformRecovery0000: Default PlatformRecovery               0x0001
[Bds]=============End Load Options Dumping=============
[Bds]BdsWait ...Zzzzzzzzzzzz...
[Bds]BdsWait(5)..Zzzz...
[Bds]BdsWait(4)..Zzzz...
[Bds]BdsWait(3)..Zzzz...
[Bds]BdsWait(2)..Zzzz...
[Bds]BdsWait(1)..Zzzz...
[Bds]Exit the waiting!
```
 After that I'm thrown into UEFI shell. I can type `exit`, which lets me choose the boot option, but when I choose `UEFI UIApp` I end up back in UEFI shell with a `map: No mapping found.` message.
So entering Tianocore setup isn't something I can do right now.